File: typescript/structural/composite/graphic/dot.ts
import {Graphic} from "./graphic";

export class Dot implements Graphic {
  x: number
  y: number

  constructor(x: number, y: number) {
    this.x = x
    this.y = y
  }

  draw(): void {
    console.log(`DOT: x = ${this.x}, y = ${this.y}`)
  }

  move(x: number, y: number): void {
    this.x += x
    this.y += y
    this.draw()
  }
}

File: dart/structural/composite/graphic/bin/dot.dart
import './graphic.dart';

class Dot implements Graphic {
  int x;
  int y;

  Dot(int xValue, int yValue) {
    x = xValue;
    y = yValue;
  }

  @override
  void draw() {
    print('DOT: x = $x, y = $y');
  }

  @override
  void move(int xValue, int yValue) {
    x += xValue;
    y += yValue;
  }
}
